Subject: reboot when panic_timout is set
From: Olaf Hering <olh@suse.de>
Patch-mainline: 2.6.19

    [POWERPC] reboot when panic_timout is set
    
    Only call into RTAS when booted with panic=0 because the RTAS call
    does not return.  The system has to be rebooted via the HMC or via the
    management console right now.  This is cumbersome and not what the
    default panic=180 is supposed to do.
    
    Signed-off-by: Olaf Hering <olh@suse.de>
    Signed-off-by: Paul Mackerras <paulus@samba.org>

diff --git a/arch/powerpc/kernel/rtas.c b/arch/powerpc/kernel/rtas.c
index 6b0699b..6ef80d4 100644
--- a/arch/powerpc/kernel/rtas.c
+++ b/arch/powerpc/kernel/rtas.c
@@ -628,6 +628,9 @@ void rtas_os_term(char *str)
 {
 	int status;
 
+	if (panic_timeout)
+		return;
+
 	if (RTAS_UNKNOWN_SERVICE == rtas_token("ibm,os-term"))
 		return;
 
